### ISSUE TITLE
1.10.0 on Windows requires UAC to launch

### ISSUE THREAD

**Issue by byjrack** [rancher-sandbox/rancher-desktop#5692]
### Actual Behavior

UI never displays and all processes close

### Steps to Reproduce

Fresh install of 1.10.0 using an Elevated prompt
Launch as a User context

### Result

Nothing in UI

background.log 
2023-10-10T16:11:59.269Z: Stopping background process Win32 socket proxy (pid <none>).
2023-10-10T16:12:55.928Z: Error starting up: Error: C:\windows\system32\wsl.exe exited with code 1
 Error: C:\windows\system32\wsl.exe exited with code 1
    at ChildProcess.<anonymous> (C:\Program Files\Rancher Desktop\resources\app.asar\dist\app\background.js:1:9554)
    at ChildProcess.emit (node:events:527:28)
    at ChildProcess._handle.onexit (node:internal/child_process:291:12)

Most other logs are empty.

### Expected Behavior

RD starts up fine as a User and uses the RDPS to handle elevated activities as needed.

### Additional Information

- I have gotten it to launch using UAC.  I did enable verbose logs though it didn't seem to make a difference in the phase it is failing at.
- wsl launches Ubuntu fine as a 2 distro.
- Already done a factory reset using rdctl as UI is inaccessible.
- Done a couple reboots as well and no change
- RD does not request UAC during Usermode startup and just exits.  I need to specify "run as admin" on launch.

### Rancher Desktop Version

1.10.0

### Rancher Desktop K8s Version

n/a

### Which container engine are you using?

moby (docker cli)

### What operating system are you using?

Windows

### Operating System / Build Version

10 Ent 19044.3448

### What CPU architecture are you using?

x64

### Linux only: what package format did you use to install Rancher Desktop?

None

### Windows User Only

standard stuff for security.  no vpn in this case.

**Comment by gunamata** [comment]
@byjrack , I could not reproduce the issue with below steps. I am on a Windows 10 Enterprise (OS Build: 19045.3448) machine. Am I doing something different from what you tried, intended?

- Open a command terminal (`cmd`) as an administrator
- Run the Rancher Desktop MSI installer from the terminal above and go through the usual installation wizard steps. 
- Launch Rancher Desktop by double clicking on the desktop shortcut. I also tried launching the app from a regular/non-admin cmd terminal.

This steps launched Rancher Desktop with out issues for me.

**Comment by byjrack** [comment]
That's it and agreed something environmental.  Have a couple users in the same situation so going to drop back to 1.9 and see what happens on the same machine.  Never had that issue before on previous releases.  Also going to try to build from source again (had a previous bug that was environmental a bit back) and try to see exactly what is going on.

Would love if the logs had more to go off of, but even with debug on I get almost nothing in the logs prior to rancher closing it's processes.  The wsl.exe error 1 seems to trigger the close, but why it gets the 1 is the unknown.

**Comment by byjrack** [comment]
1.9.1 has the same result, but I noticed it had more in it's background log

```
2023-10-11T11:30:57.384Z: Error starting up: c [Error]: C:\windows\system32\wsl.exe exited with code 1
    at ChildProcess.<anonymous> (C:\Program Files\Rancher Desktop\resources\app.asar\dist\app\background.js:1:15549)
    at ChildProcess.emit (node:events:527:28)
    at ChildProcess._handle.onexit (node:internal/child_process:291:12) {
  command: [
    'C:\\windows\\system32\\wsl.exe',
    '--distribution',
    'wsl-vpnkit',
    '--exec',
    '/bin/wslpath',
    '-a',
    '-u',
    'C:\\Program Files\\Rancher Desktop\\resources\\resources\\linux\\bin\\docker-buildx'
  ],
  stdout: '',
  code: 1,
  [Symbol(child-process.command)]: 'C:\\windows\\system32\\wsl.exe --distribution wsl-vpnkit --exec /bin/wslpath -a -u C:\\Program Files\\Rancher Desktop\\resources\\resources\\linux\\bin\\docker-buildx'
}
```
For our non-RD distros on VPN clients we have been using wsl-vpnkit with the systemd binding.  I know RD has it's own gvisor support which in this case is disabled for this install as it wasn't needed.  Now that integration is done in the distro, but I was wondering in the wsl-helper integration prep if the vpnkit distro confused RD and it bailed.  So I unregistered wsl-vpnkit and 1.9.1 launched as expected.  

I re-registed wsl-vpnkit and things were back to broken with the same wslpath error in background.log.  I did notice that the path to docker-buildx was not enclosed in quotes in the log.  That definitely wouldn't work given install to PF.  I can see a single line in wsl-helper.wsl-vpnkit.log.

`/bin/wslpath: C:\Program Files\Rancher Desktop\resources\resources\linux\bin\docker-buildx`

Which makes me think https://github.com/rancher-sandbox/rancher-desktop/blob/main/pkg/rancher-desktop/integrations/windowsIntegrationManager.ts#L217 when calling wsl it needs to enclose that in quotes.  Going to try and hack that in and test.

Now why this only happens on wsl-vpnkit and why UAC matters here is a head scratcher, but I am still digging.

**Comment by jandubois** [comment]
Oh, it is the wsl-vpnkit issue #4966; that has been fixed for 1.11 in #4560.

Can you confirm with a CI build and close this issue if it is indeed fixed by current `main` builds?

**Comment by byjrack** [comment]
> Oh, it is the wsl-vpnkit issue #4966; that has been fixed for 1.11 in #4560.
> 
> Can you confirm with a CI build and close this issue if it is indeed fixed by current `main` builds?

Right on the money.  Sorry did a search initially on UAC/admin requirements and no hits, but after tracking back to the distro integration I didn't swing back through Issues to look for hits.  #4966 was the exact.

Did an install of [292](https://github.com/rancher-sandbox/rancher-desktop/actions/runs/6485399854) and no issues during startup with wsl-vpnkit registered.  Calling this one closed and will keep an eye out for 1.11 for the official.
